routes/course.js

```diff
@@ -262,4 +262,83 @@ router.delete("/api/courses/:course_id", auth, async (req, res) => {
   }
 });
 
+
+
+// delete and create a new course
+// need to provide new course as json object in request body
+// distribution field is also updated
+router.patch("/api/courses/:course_id", auth, async (req, res) => {
+  // delete the course
+  const c_id = req.params.course_id;
+  try {
+    // verify that course belongs to req user
+    const course = await Courses.findById(c_id).exec();
+    if (!course) {
+      return errorHandler(res, 404, "Course not found."); 
+    } else if (req.user._id !== course.user_id) {
+      return forbiddenHandler(res);
+    }
+    // delete course and update distributions
+    await Courses.findByIdAndDelete(c_id).exec();
+    await Years
+      .findByIdAndUpdate(
+        course.year_id,
+        { $pull: { courses: course._id }},
+        { runValidators: true }
+      )
+      .exec();
+    for (let id of course.distribution_ids) {
+      const distribution = await Distributions
+        .findByIdAndUpdate(
+          id,
+          { $pull: { courses: c_id } },
+          { new: true, runValidators: true }
+        )
+        .exec();
+      await distributionCreditUpdate(distribution, course, false);
+    }
+    returnData(course, res);
+  } catch (err) {
+    errorHandler(res, 500, err);
+  }
+
+  // insert a new course
+  const course = req.body.new_course;
+  if (!course || Object.keys(course).length == 0) {
+    return missingHandler(res, { course });
+  }
+  if (course.user_id !== req.user._id) {
+    return forbiddenHandler(res);
+  }
+  try {
+    // check that course distributions belong to plan
+    const plan = await Plans.findById(course.plan_id).exec();
+    course.distribution_ids.forEach((id) => {
+      if (!plan.distribution_ids.includes(id))
+        errorHandler(res, 500, {
+          message: "Invalid combination of plan_id and distribution_ids.",
+        });
+    });
+    const retrievedCourses = await Courses.findByPlanId(course.plan_id); 
+    for (const existingCourse of retrievedCourses) {
+      if(existingCourse.number === course.number && existingCourse.term === course.term && existingCourse.year === course.year && existingCourse.title === course.title){
+        return errorHandler(res, 400, { 
+          message: "Cannot take same course multiple times in the same semester",
+        });
+      }
+    }
```

Comment:
I think this would not be necessary as hopefully there were no duplicate courses to begin with. 
but maybe that's not a correct assumption. 

I'll leave it up to you!